core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+      await this.clickOn(addTags);
+      await this.type(addTags, TagNames[i].toLowerCase());
+      await this.page.keyboard.press('Tab');
+    }
+  }
+
+  /**
+   * This function checks if tags are successfully added.
+   */
+  async expectTagsToBeAdded(expectedTags: string[]): Promise<void> {
+    const lowercaseExpectedTags = expectedTags.map(tag => tag.toLowerCase());
+    await this.page.waitForSelector('mat-chip-list');
+
+    const addedTags = await this.page.evaluate(() => {
+      const tagElements = Array.from(document.querySelectorAll('mat-chip'));
+      return tagElements
+        .map(tag => {
+          const textContent =
+            tag.querySelector('.mat-chip-remove')?.previousSibling?.textContent;
+          return textContent ? textContent.trim() : '';
+        })
+        .filter(Boolean);
+    });
+
+    for (const expectedTag of lowercaseExpectedTags) {
+      if (!addedTags.includes(expectedTag)) {
+        throw new Error(`Tag "${expectedTag}" was not added.`);
+      }
+    }
+
+    showMessage('All expected tags were added successfully.');
+  }
+
+  /**
+   * This function allows you to preview the exploration.
+   */
+  async previewSummary(): Promise<void> {
+    await this.page.waitForSelector(
+      '.e2e-test-open-preview-summary-modal:not([disabled])'
+    );
+    await this.clickOn(previewSummaryButton);
+    await this.clickOn(dismissPreviewButton);
+    await this.expectPreviewSummaryToBeVisible();
+  }
+
+  /**
+   * This function verifies that the preview summary is visible.
+   */
+  async expectPreviewSummaryToBeVisible(): Promise<void> {
+    await this.page.waitForSelector('.e2e-test-open-preview-summary-modal');
+    const previewSummary = await this.page.$(
+      '.e2e-test-open-preview-summary-modal'
+    );
+
+    if (previewSummary) {
+      showMessage('Preview summary is visible.');
+    } else {
+      throw new Error('Preview summary is not visible.');
+    }
+  }
+
+  /**
+   * This function helps in updating advanced settings
+   */
+  async enableAutomaticTextToSpeech(): Promise<void> {
+    await this.clickOn(textToSpeechToggle);
+  }
+
+  /**
+   * This function checks whether the Automatic Text-to-Speech
+   * setting is enabled or disabled.
+   */
+  async expectAutomaticTextToSpeechToBeEnabled(): Promise<void> {
+    await this.page.waitForSelector('#text-speech-switch');
+    const autoTTSwitch = await this.page.$('#text-speech-switch');
+    const isAutoTTSwitchOn = await this.page.evaluate(
+      switchElement => switchElement.checked,
+      autoTTSwitch
+    );
+    if (isAutoTTSwitchOn) {
+      showMessage('Automatic Text-to-Speech is enabled.');
+    } else {
+      throw error('Automatic Text-to-Speech is disabled.');
+    }
+  }
+
+  /**
+   * This function helps in assigning role of collaborator to any guest user.
+   */
+  async assignUserToCollaboratorRole(username: string): Promise<void> {
+    await this.clickOn(editbutton);
```

Comment:
editButton -- though you should make this more specific, what is the button editing?

Name the button so that it's obvious to someone looking at the UI what the button is (if they're asked to find it). Don't use ambiguous names.